From 165505c64fa2d3a0694e1e9239aa20a5d8552f16 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Mon, 13 Feb 2012 11:07:15 +0800
Subject: [PATCH 195/256] ENGR00174040 vpu: sync vpu lib to align with v2.1.3
 f/w on i.mx6x

Sync vpu lib to align with v2.1.3 f/w on i.mx6x platforms.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c  |  2 ++
 vpu/vpu_reg.h  | 21 ++++++++++++++++++---
 vpu/vpu_util.h |  3 ++-
 3 files changed, 22 insertions(+), 4 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 53f92c7..aeca301 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -537,6 +537,8 @@ RetCode vpu_EncOpen(EncHandle * pHandle, EncOpenParam * pop)
 	pEncInfo = &pCodecInst->CodecInfo.encInfo;
 
 	pCodecInst->contextBufMem.size = SIZE_CONTEXT_BUF;
+	if (cpu_is_mx6x() && pop->bitstreamFormat == STD_AVC)
+		pCodecInst->contextBufMem.size += PS_SAVE_SIZE;
 	ret = IOGetPhyMem(&pCodecInst->contextBufMem);
 	if (ret) {
 		err_msg("Unable to obtain physical mem\n");
diff --git a/vpu/vpu_reg.h b/vpu/vpu_reg.h
index 7b21f66..e0e87bc 100644
--- a/vpu/vpu_reg.h
+++ b/vpu/vpu_reg.h
@@ -237,7 +237,7 @@
 #define CMD_DEC_SEQ_USER_DATA_OPTION	0x194
 #define CMD_DEC_PIC_USER_DATA_BASE_ADDR 0x1AC
 #define CMD_DEC_PIC_USER_DATA_BUF_SIZE  0x1B0
-#define CMD_DEC_PIC_DBK_OFFSET		0x1B4
+#define CMD_DEC_PIC_NUM_ROWS		0x1B4
 
 #define CMD_DEC_PIC_THO_PIC_PARA	0x198
 #define CMD_DEC_PIC_THO_QMAT_ADDR	0x1A0
@@ -269,6 +269,7 @@
 
 #define RET_DEC_PIC_CROP_LEFT_RIGHT	0x1E0
 #define RET_DEC_PIC_CROP_TOP_BOTTOM	0x1E4
+#define RET_DEC_PIC_RV_TR		0x1E8
 #define RET_DEC_PIC_VP8_PIC_REPORT	0x1E8
 #define RET_DEC_PIC_ASPECT		0x1F0
 #define RET_DEC_PIC_VP8_SCALE_INFO	0x1F0
@@ -396,9 +397,9 @@
         #define PARA_BUF2_SIZE			(2 * 1024)
         #define PARA_BUF_SIZE			(10 * 1024)
 #elif defined(IMX6Q)
-	#define CODE_BUF_SIZE                   (240 * 1024) /* define max is 240k byte currently */
+	#define CODE_BUF_SIZE                   (280 * 1024) /* define max is 280k byte currently */
 	#define FMO_SLICE_SAVE_BUF_SIZE         (32)
-	#define TEMP_BUF_SIZE                   (200 * 1024)
+	#define TEMP_BUF_SIZE                   (204 * 1024)
 	#define PARA_BUF2_SIZE                  (2 * 1024)
 	#define PARA_BUF_SIZE                   (10 * 1024)
 #else
@@ -482,6 +483,20 @@
 #define GDI_BUS_STATUS              (GDMA_BASE + 0x0F4)
 
 #define GDI_SIZE_ERR_FLAG           (GDMA_BASE + 0x0E0)
+#define GDI_ADR_RQ_SIZE_ERR_PRI0    (GDMA_BASE + 0x100)
+#define GDI_ADR_RQ_SIZE_ERR_PRI1    (GDMA_BASE + 0x104)
+#define GDI_ADR_RQ_SIZE_ERR_PRI1    (GDMA_BASE + 0x104)
+#define GDI_ADR_RQ_SIZE_ERR_PRI2    (GDMA_BASE + 0x108)
+#define GDI_ADR_WQ_SIZE_ERR_PRI0    (GDMA_BASE + 0x10c)
+#define GDI_ADR_WQ_SIZE_ERR_PRI1    (GDMA_BASE + 0x110)
+#define GDI_ADR_WQ_SIZE_ERR_PRI2    (GDMA_BASE + 0x114)
+#define GDI_ADR_RQ_SIZE_ERR_SEC0    (GDMA_BASE + 0x118)
+#define GDI_ADR_RQ_SIZE_ERR_SEC1    (GDMA_BASE + 0x11c)
+#define GDI_ADR_RQ_SIZE_ERR_SEC2    (GDMA_BASE + 0x120)
+#define GDI_ADR_WQ_SIZE_ERR_SEC0    (GDMA_BASE + 0x124)
+#define GDI_ADR_WQ_SIZE_ERR_SEC1    (GDMA_BASE + 0x128)
+#define GDI_ADR_WQ_SIZE_ERR_SEC2    (GDMA_BASE + 0x12c)
+
 #define GDI_INFO_CONTROL            (GDMA_BASE + 0x400)
 #define GDI_INFO_PIC_SIZE           (GDMA_BASE + 0x404)
 #define GDI_INFO_BASE_Y             (GDMA_BASE + 0x408)
diff --git a/vpu/vpu_util.h b/vpu/vpu_util.h
index 193ead2..d511087 100644
--- a/vpu/vpu_util.h
+++ b/vpu/vpu_util.h
@@ -44,11 +44,12 @@ typedef enum {
 
 
 #if defined(IMX6Q)
-#define BIT_WORK_SIZE			47 * 1024
+#define BIT_WORK_SIZE			48 * 1024
 #else
 #define BIT_WORK_SIZE			128 * 1024
 #endif
 #define SIZE_CONTEXT_BUF		BIT_WORK_SIZE
+#define PS_SAVE_SIZE			0x080000
 
 #define SIZE_PIC_PARA_BASE_BUF          0x100
 #define SIZE_MV_DATA                    0x20000
-- 
1.8.0

